### ISSUE TITLE
Smokey errors on pages with embedded YouTube video

### ISSUE THREAD

**Issue by whoojemaflip** [alphagov/smokey#279]
```
One or more errors were raised in the Javascript code on the page. If you don't care about these errors, you can ignore them by setting js_errors: false in your Poltergeist configuration (see documentation for details).

TypeError: 'undefined' is not a function (evaluating 'this.g.pause()')
    at https://www.youtube.com/yts/jsbin/player-vflrAeTuj/en_GB/base.js:6447
    at https://www.youtube.com/yts/jsbin/player-vflrAeTuj/en_GB/base.js:3388
    at https://www.youtube.com/yts/jsbin/player-vflrAeTuj/en_GB/base.js:3363
    at https://www.youtube.com/yts/jsbin/player-vflrAeTuj/en_GB/base.js:4119
    at https://www.youtube.com/yts/jsbin/player-vflrAeTuj/en_GB/base.js:4098
    at https://www.youtube.com/yts/jsbin/player-vflrAeTuj/en_GB/base.js:7193
    at https://www.youtube.com/yts/jsbin/www-embed-player-vfl8435pg/www-embed-player.js:383
    at https://www.youtube.com/yts/jsbin/www-embed-player-vfl8435pg/www-embed-player.js:23
    at https://www.youtube.com/yts/jsbin/www-embed-player-vfl8435pg/www-embed-player.js:384
    at Rg (https://www.youtube.com/yts/jsbin/www-embed-player-vfl8435pg/www-embed-player.js:381)
    at https://www.youtube.com/yts/jsbin/www-embed-player-vfl8435pg/www-embed-player.js:373
    at https://www.youtube.com/yts/jsbin/www-embed-player-vfl8435pg/www-embed-player.js:421

(Capybara::Poltergeist::JavascriptError)
./features/support/visiting_pages.rb:11:in `/^I visit "(.*)"$/'
features/design_principles.feature:15:in `When I visit "/transformation"'

Failing Scenarios:
cucumber -p production features/design_principles.feature:14 # Scenario: check Transformation dashboard
```

**Comment by andrewgarner** [comment]
I've been investigating this, I'm seeing the failure intermittently. Here is a small script that replicated the error:

```ruby
require 'bundler/setup'

require 'capybara/dsl'
require 'capybara/poltergeist'

Capybara.app_host = 'https://www.gov.uk/'
Capybara.current_driver = :poltergeist

include Capybara::DSL

visit '/transformation'
```

I've tried bumping the dependencies of Capybara and Poltergeist but no joy.

I've tried directly with PhantomJS and don't get the error:

```js
var page = require('webpage').create();
page.open('https://www.gov.uk/transformation', function(status) {
  console.log("Status: " + status);
  if(status === "success") {
    page.render('debug.png');
  }
  phantom.exit();
});
```

**Comment by andrewgarner** [comment]
Perhaps this is related – https://github.com/ariya/phantomjs/issues/10839

Could the failure occur if the async HTML5 video code loads before Poltergeist has done it's thing? Perhaps the YouTube embed code has changed causing the failures to start surfacing?

**Comment by chrisroos** [comment]
@andrewgarner: I've got a relatively simple "test" that allows me to reproduce the problem in this repo - https://github.com/chrisroos/phantomjs-and-youtube-embeds. I plan to continue investigating tomorrow.

**Comment by chrisroos** [comment]
I've opened PR #280 that adds youtube.com to Poltergeist's blacklist for our Smokey tests. This fixes the intermittent failures we're seeing on pages with embedded YouTube videos but doesn't get to the bottom of what's actually causing the problem.

**Comment by chrisroos** [comment]
I've now merged PR #280 and it's been deployed to integration. I kicked the Smokey build on integration and it passed: https://deploy.integration.publishing.service.gov.uk/job/Smokey/18729/. We should keep an eye on it but my hope is that we should no longer see these YouTube errors.

I'm going to close this issue but we should reopen it if we discover that the problem reappears.

**Comment by nickcolley** [comment]
Nice work @chrisroos 👍
